tests/unit/src/flixel/FlxObjectTest.hx

```diff
@@ -74,58 +74,72 @@ class FlxObjectTest extends FlxTest
 		Assert.isFalse(FlxG.overlap(object1, object2));
 	}
 	
+	@Test // closes #1564, tests #1561
+	function testSeparateYAfterX():Void
+	{
+		var object1:FlxObject = new FlxObject(0, 0, 15, 15);
+		var bar:FlxObject = new FlxObject(17, 17, 100, 40);
+		bar.immovable = true;
+		FlxG.state.add(object1);
+		FlxG.state.add(bar);
+		object1.velocity.set(100, 100);
+		step();
+		Assert.isTrue(FlxG.collide(object1, bar));
+		Assert.areEqual(bar.y, object1.y + object1.height);
+	}
+
 	@Test
 	function testUpdateTouchingFlagsHorizontal():Void
 	{
-		var object1 = new FlxObject(0, 0, 10, 10);
-		var object2 = new FlxObject(20, 2, 10, 6);
+		var object1:FlxObject = new FlxObject(0, 0, 10, 10);
```

Comment:
Not a fan of adding the `:FlxObject` explicitly here. That's where Haxe's type inference comes in handy and helps to avoid redundancy (see in the "Leveraging Type Inference" section [here](http://haxeflixel.com/documentation/code-style/)).